pkg/microservice/aslan/core/common/repository/mongodb/build.go

```diff
@@ -166,6 +172,44 @@ func (c *BuildColl) List(opt *BuildListOption) ([]*models.Build, error) {
 	return resp, nil
 }
 
+func (c *BuildColl) ListByOptions(opt *BuildListOption) ([]*models.Build, int64, error) {
+	if opt == nil {
+		return nil, 0, errors.New("nil ListOption")
+	}
+
+	query := bson.M{}
+	if len(opt.ProductName) != 0 {
+		query["product_name"] = opt.ProductName
+	}
+
+	var resp []*models.Build
+	ctx := context.Background()
+	opts := options.Find()
+	if opt.IsSort {
+		opts.SetSort(bson.D{{"update_time", -1}})
+	}
+	if opt.PageNum > 0 && opt.PageSize > 0 {
+		opts.SetSkip((opt.PageNum - 1) * opt.PageSize)
+		opts.SetLimit(opt.PageSize)
+	}
+
+	count, err := c.Collection.CountDocuments(ctx, query)
+	if err != nil {
+		return nil, 0, err
+	}
+	cursor, err := c.Collection.Find(ctx, query, opts)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	err = cursor.All(ctx, &resp)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return resp, count, nil
+}
+
```

Comment:
May improve this paging function to only one mongo query

Reply:
optimize later here, not much impact here at the moment.